### DISCUSSION TITLE
VdhCoApp not recognized

### DISCUSSION THREAD

**Discussion by Paolo59it** [aclap-dev/video-downloadhelper#10] [Help and Q&A]
### Your question

Hello,
I had this issue some weeks ago and solved it with the help at
https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized

Today, after a Firefox update (snap install on Ubuntu 22.04 64bit, browser ver snap 122.0) the issue is up again. The add-on is not able to find the companion app. I tried what worked last time, to no avail. I ran:
sudo apt-get install -y flatpak flatpak permission-set webextensions net.downloadhelper.coapp snap.firefox yes
and
/opt/vdhcoapp/vdhcoapp install
No error occured.

Link to VdhCoApp installed:
https://github.com/aclap-dev/vdhcoapp/releases/download/v2.0.10/vdhcoapp-2.0.10-linux-x86_64.deb
Output of /opt/vdhcoapp/vdhcoapp --info:
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "/opt/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/paolo",
  "converterBinary": "/opt/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

Thank you for any help.

Paolo

### Addon & CoApp version and details

version: 8.1.4.0
beta: false
build date: 2024-01-04
build options: linuxlic: false, noyt: false, browser: firefox
lang: it
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: no.
(I have coapp, the addon in Chrome correctly sees it.)

**Comment by paulrouget** [comment]
Did you run these commands as your regular user (never run these commands as root):

```shell
/opt/vdhcoapp/vdhcoapp install
flatpak permission-set webextensions net.downloadhelper.coapp snap.firefox yes
```

They are meant to be run without sudo.
Make sure to restart Firefox after that.

**Comment by paulrouget** [comment]
Ok so I can reproduce here. Firefox Snap version 122 doesn't recognize the coapp.

Working on it.

**Comment by paulrouget** [comment]
I'm closing this in favor of https://github.com/aclap-dev/video-downloadhelper/discussions/12
